src/transaction_builder.cpp

```diff
@@ -20,6 +20,31 @@ SpendDescriptionInfo::SpendDescriptionInfo(
     librustzcash_sapling_generate_r(alpha.begin());
 }
 
+TransactionBuilderResult::TransactionBuilderResult(const CTransaction& tx) : maybeTx(tx) {}
+
+TransactionBuilderResult::TransactionBuilderResult(const std::string& error) : maybeError(error) {}
+
+bool TransactionBuilderResult::isTx() { return maybeTx != boost::none; }
+
+bool TransactionBuilderResult::isError() { return maybeError != boost::none; }
+
+CTransaction TransactionBuilderResult::getTxOrThrow() {
+    if (maybeTx) {
+        return maybeTx.get();
+    } else {
+        throw std::runtime_error(getError());
+    }
+}
+
+std::string TransactionBuilderResult::getError() {
+    if (maybeError) {
+        return maybeError.get();
+    } else {
+        // This can only happen if isTx() is true in which case we should not call getError()
+        return "Unknown Error";
```

Comment:
The irony of this is that this error can only occur if building the transaction was successful but then we try to get the error nonetheless. What about changing this error to be `"Cannot return an error from a successfully built transaction"`. If a user ever sees this message it means that we have a bug with a simple fix. It could also say `"Do not call getError() without first calling isTx() or isError()"`. I think I actually like the second option better because if we make this mistake it tells us how to fix it.

Reply:
I wouldn't use the first phrasing - cause we just know `Build` returned a CTransaction - but maybe it is not a well-built tx for some reason we've overlooked. Yeah the second is better, don't know if perfect, how about `getError() called in TransactionBuilderResult although result isn't initialized as error`,

Reply:
I like your suggestion, so I will do approximately that. I am also changing it to just throw that as a runtime error - I think this is consistent with the idea that we should not be calling this method if we do not have an error to get, and ultimately it will/should just end up getting thrown anyway.